storage/src/hash/mod.rs

```diff
@@ -0,0 +1,444 @@
+// Copyright (c) SimpleStaking and Tezedge Contributors
+// SPDX-License-Identifier: MIT
+
+mod ocaml;
+
+use std::{array::TryFromSliceError, convert::TryInto, io};
+
+use blake2::digest::{InvalidOutputSize, Update, VariableOutput};
+use blake2::VarBlake2b;
+use failure::Fail;
+use ocaml::ocaml_hash_string;
+
+use crate::merkle_storage::{Commit, ContextValue, Entry, Node, NodeKind, Tree};
```

Comment:
@tizoc 
we need to clean this little bit more later,
but for this PR, could you please move this hash dir to another subdir `context`, to look like:
```
storage/src/context/hash/mod.rs 
```

later with Mateusz PR with GC, we will move: `context.rs` and `merkle_storage.rs` also to `storage/src/context`,
and those common stuff: `use crate::merkle_storage::{Commit, ContextValue, Entry, Node, NodeKind, Tree};` we could move for example to: `storage/src/context/mod.rs`

and very later, my intention is to split the whole storage library to smaller dedicated libraries - I guess this will also simplify ffi context integration - we will see later

Reply:
Sounds good, something that I find very smelly right now is how `hash` depends on stuff from the upper level, which also depends on stuff from `hash`, makes me uncomfortable.

Reply:
exactly, looks like kind of circular dependency now, thats why I want to move common stuff to src/context/mod.rs later